tests/correctness/control_flow.rs

```diff
@@ -483,6 +483,61 @@ fn for_loop_and_increment_10_times_skipping_1() {
     assert_eq!(res, 1005);
 }
 
+#[test]
+fn for_loop_does_not_execute_if_condition_is_not_met() {
+    let function = r#"
+    FUNCTION main : DINT
+    VAR
+        i, end, step, res: DINT;
+    END_VAR
+        end := -1;
+        step := -1;
+        res := 100;
+
+        FOR i := end TO 0 BY step DO
+            res := i;
+        END_FOR
+
+        main := res;
+    END_FUNCTION
+    "#;
+
+    let res: i32 = compile_and_run(function.to_string(), &mut crate::MainType::default());
+    assert_eq!(res, 100);
+}
+
+#[test]
+fn for_loop_step_changes_sign_in_loop_body() {
+    let function = r#"
+    FUNCTION main : DINT
+    VAR
+        i, step, temp, iteration: DINT;
+    END_VAR
+        step := 1;
+        iteration := 0;
+
+        FOR i := 5 TO 10 BY step DO
+            temp := (step + 1) * -1 ;
+            IF i + temp > 0 THEN
+                step := temp;
+            ELSE
+                step := step + 3;
+            END_IF;
+
+            // i:     5, 3,  4, 2,  3, 1, 2,  6, 1, 5, 12
+            // step: -2, 1, -2, 1, -2, 1, 4, -5, 4, 7
+            // we expect 10 iterations
+            iteration := iteration + 1;
+        END_FOR
+        
+        main := iteration;
+    END_FUNCTION
```

Comment:
I've just thought of a weird edge case, where the signage of the `STEP` variable changes in the loop body. What behaviour do we expect here? do we loop until `COUNTER >= END` or do we also exit the loop if `COUNTER <= START` if we initially determine the loop to be incrementing?